src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -73,50 +82,100 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
         }
 
         public void AddNew() {
-            SelectConnection(new ConnectionViewModel());
+            _shell.AssertIsOnMainThread();
+            CancelEdit(SelectedConnection);
+            NewConnection = new ConnectionViewModel();
+        }
+        
+        public void CancelEdit(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            if (connection == NewConnection) {
+                NewConnection = null;
+            } else if (connection != null) {
+                connection.Reset();
```

Comment:
Agreed.

---
In reply to: [77055531](https://github.com/Microsoft/RTVS/pull/2237#discussion_r77055531) [](ancestors = 77055531)